### PR TITLE
Tm 6687 access smtp lib to send emails

### PR DESCRIPTION
**Pull request by mansi-mp-200**
[Ticket][(linkToTicket)](https://metaphase.atlassian.net/jira/software/c/projects/TM/boards/1?assignee=712020%3A3d9639a5-439c-4e2d-9bce-37b6924384d2&selectedIssue=TM-6687)

Adding functionality to send a test email to DosDev SMTP server. This endpoint will be used to send TM1 emails for notifications and memos for the work Jeanie is doing. 

This is just a basic function to show the use of smtplib and how we can connect to the DosDev servers. More notable work needs to be done, specifically adding attachment functionality as well as not hardcoding server information but bringing that in the form of environment variables from a secured location. 

SS of emails received in inbox:

SMTP Test Email 1

SMTP Email 1 Morgan

How to test in Postman:

1. 2 variables needed for Header: Jwt (your JWT token) and Authorization (found as "Token ...") - both can be found when using the Network tab in Chrome Dev Tools when looking up the UI
2. The following data keys needs to be send as part of the body and can be send as form-data. Example:

Sending Emails TM1 postman body

### CODE REVIEW

File: talentmap_api/fsbid/views/notifications.py [hunk 1]
```diff
@@ -273,3 +273,75 @@ def put(self, request):
...
+        
+
+class SendTM1EmailView(APIView):
+    """
+    Sends TM1 emails via SMTP using service emails. 
+
+    Only basic template. More work needs to be done for recipients, and attachment issues as well as 
+    pulling in Environment variables from a more secure location.
+    """
+    permission_classes = [IsAuthenticatedOrReadOnly, Or(And(isDjangoGroupMember('superuser'), isDjangoGroupMember('fsbid_admin')),
```
Comment (jlam-mpc) [inline talentmap_api/fsbid/views/notifications.py:311]: I think we might have to add ELCDO too but that can probably be saved for later when we have to make a future ticket to handle that entire role. I'm unsure but if its not listed in your ticket you can ignore this note.

File: talentmap_api/fsbid/views/notifications.py [hunk 2]
```diff
@@ -273,3 +273,75 @@ def put(self, request):
...
+            type=openapi.TYPE_OBJECT,
+            properties={
+                'to': openapi.Schema(type=openapi.TYPE_ARRAY, items=openapi.Items(type=openapi.TYPE_STRING), description='List of recipient email addresses'),
+                'subject': openapi.Schema(type=openapi.TYPE_STRING, description='Email subject'),
+                'body': openapi.Schema(type=openapi.TYPE_STRING, description='Email body content'),
+            },
+            example={
+                'to': ['recipient1@example.com', 'recipient2@example.com'],
+                'subject': 'Your Subject Here',
+                'body': 'Your email body content here',
```
Comment (jlam-mpc) [inline talentmap_api/fsbid/views/notifications.py:325]: Just confirming that it's only to pass in an empty string here, right?

File: talentmap_api/fsbid/views/notifications.py [hunk 3]
```diff
@@ -273,3 +273,75 @@ def put(self, request):
         '''
         result = services.send_to_eopf(request.data, request.META['HTTP_JWT'])
         return Response(result)
+
+
+class GetEmailFromRequest(APIView):
```
Comment (jlam-mpc) [inline talentmap_api/fsbid/views/notifications.py:278]: Really nice logic here

File: talentmap_api/fsbid/views/notifications.py [hunk 4]
```diff
@@ -273,3 +273,75 @@ def put(self, request):
...
+    """
+    Returns email of the logged in user. In Dev1, emails can only be send from the talentmap@elguaria.net email so
```
Comment (jlam-mpc) [inline talentmap_api/fsbid/views/notifications.py:280]: Might be good to note this in either yours or my tickets so testers know about this went testing in DEV1

I'm very impressed and appreciative of your documented code this is wonderful

**Review by jlam-mpc — APPROVED** [review]
Super well documented and the logic looks good

I can't test this locally but it's readable and makes sense and satisfies your ticket so go ahead and merge

I just left a few notes and questions for myself

File: talentmap_api/fsbid/views/notifications.py (hunk 2 above)
Comment (mansi-mp-200) [reply talentmap_api/fsbid/views/notifications.py:325]: I'm sorry what do you mean by this comment? Pass in an empty string where? The body?

File: talentmap_api/fsbid/views/notifications.py (hunk 4 above)
Comment (mansi-mp-200) [reply talentmap_api/fsbid/views/notifications.py:280]: Ah yes I mention that this goes with my work. 

Thank you :)

**Pull request merged**